redwoods-insurance-salesforce/force-app/main/default/lwc/incidentAudioPlayer/__tests__/incidentAudioPlayer.test.js

```diff
@@ -0,0 +1,100 @@
+import { getAudio } from '@salesforce/apex/IncidentController.findRelatedFiles';
+
+const mockAudioFile = require('./data/mockAudioFile.json');
+const multipleMockAudioFiles = require('./data/multipleMockAudioFiles.json');
+const getRelatedAudioAdapter = registerApexTestWireAdapter(getAudio);
+
+describe('c-incident-audio-player', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('invokes the wire adapter with default properties', () => {
+        // Create initial element
+        const element = createElement('c-incident-audio-player', {
+            is: IncidentAudioPlayer
+        });
+        element.recordId = 'mockRecordId';
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            expect(getRelatedAudioAdapter.getLastConfig()).toEqual({
+                caseId: 'mockRecordId',
+                fileType: 'AUDIO'
+            });
+        });
+    });
+
+    it('does not show the player when no audio file is provided.', () => {
+        // Create initial element
+        const element = createElement('c-incident-audio-player', {
+            is: IncidentAudioPlayer
+        });
+        document.body.appendChild(element);
+
+        const labelElement = element.shadowRoot.querySelector('p');
+        expect(labelElement.textContent).toContain(
+            'There are currently no audio files for this case.'
+        );
+    });
+
+    it('displays the audio player when the wire adapter returns an audio file', () => {
+        const element = createElement('c-incident-audio-player', {
+            is: IncidentAudioPlayer
+        });
+        document.body.appendChild(element);
+
+        getRelatedAudioAdapter.emit(mockAudioFile);
+
+        return Promise.resolve().then(() => {
+            const audioPlayerEl = element.shadowRoot.querySelector('audio');
+            expect(audioPlayerEl).not.toBe(null);
+            expect.stringMatching(audioPlayerEl.src, /mockAudioFile.Id$/);
+        });
+    });
+
+    it('displays one audio player for each file the wire adapter returns', () => {
+        const element = createElement('c-incident-audio-player', {
+            is: IncidentAudioPlayer
+        });
+        document.body.appendChild(element);
+
+        getRelatedAudioAdapter.emit(multipleMockAudioFiles);
+
+        return Promise.resolve().then(() => {
+            const audioPlayerEls = element.shadowRoot.querySelectorAll('audio');
+            expect(audioPlayerEls.length).toEqual(
+                multipleMockAudioFiles.length
+            );
+            audioPlayerEls.forEach(function (value, index) {
+                expect.stringMatching(
+                    audioPlayerEls[index].src,
+                    /multipeMockAudioFiles[index].Id$/
```

Comment:
Similar as with the image component, this should check for the fully expected URI, not only for the `Id`.

Reply:
Again, I disagree. The file URL is a constant with the exception of the ID. Testing the full url string violates the principle of testing only what can possibly change.

Reply:
Possible change: typo in the URL which will result in a failing component in the user experience. That why has to change to a full URL string.

Reply:
@muenzpraeger The url is hard coded except the id. You're requesting changes just to request them.

Reply:
@pozil What's your take? Testing the full URI, or just the record id?

Reply:
I'm fine with the testing just the dynamic part of the URL. We should refrain from testing static things like template elements or constants.
The url could be indeed be broken due to a typo but testing for the real URL would be an integration test, not a unit test.

Reply:
THANK YOU